File: Day 15,16/SB-EMS/src/main/java/com/coforge/ems/service/EmployeeService.java
package com.coforge.ems.service;

import java.util.List;
import java.util.Optional;

import com.coforge.ems.exception.EmployeeNotFoundException;
import com.coforge.ems.exception.InvalidEmployeeObjectException;
import com.coforge.ems.model.Employee;

public interface EmployeeService {

	public boolean saveEmployee(Employee employee) throws InvalidEmployeeObjectException;

	public boolean UpdateEmployee(int eid, Employee employee)throws InvalidEmployeeObjectException, EmployeeNotFoundException ;

	public boolean deleteEmployee(int eid)throws InvalidEmployeeObjectException, EmployeeNotFoundException;

	public Optional<Employee> findEmployee(int eid) throws EmployeeNotFoundException, InvalidEmployeeObjectException;

	public List<Employee> findAllEmployee() throws EmployeeNotFoundException, InvalidEmployeeObjectException;

	public List<Employee> FindByEname(String ename) throws InvalidEmployeeObjectException;

	public boolean deleteByEname(String ename) throws InvalidEmployeeObjectException;

	public List<Integer> getEidsList();

	public List<Object[]> getEmployeeStatistics();
}

File: Day 28/react/src/services/EmployeeService.js
import axios from "axios";


class EmployeeService  {
    constructor(){
    this.BASE_URL = "http://localhost:1111/api/v1/ems";
    }

    saveEmployee(employee) {
    return axios.post(this.BASE_URL + "/employees", employee);
    }

  updateEmployee(eid, employee) {
        return axios.put(this.BASE_URL + "/employees/" + eid, employee)
            .then((response) => {
                console.log(response.data);
                return response.data;
            });
    }

    deleteEmployee(eid) {
        return axios.delete(this.BASE_URL + "/employees/" + eid)
            .then((response) => {
                console.log(response.data);
                return response.data;
            });
    }

    findEmployee(eid) {
        return axios.get(this.BASE_URL + "/employees/" + eid)
            .then((response) => {
                console.log(response.data);
                return response.data;
            });
    }

    findAllEmployees() {
        return axios.get(this.BASE_URL + "/employees")
            .then((response) => {
                console.log(response.data);
                return response.data;
            });
    }
}

export default EmployeeService;